src/Flagged/Modal.jsx

```diff
@@ -0,0 +1,71 @@
+let {
+  onReport,
+  reportedAccountId,
+  contentModerationFlagValue,
+  ...forwardedProps
+} = props;
+
+const profileUrl = `#/${REPL_ACCOUNT}/widget/ProfilePage?accountId=${reportedAccountId}`;
+
+let getFlaggedAccountsList;
+
+// copied from Moderate.jsx so maybe it's wrong
+if (forwardedProps.open) {
+  getFlaggedAccountsList = Social.get(
+    `${context.accountId}/moderate/users`,
+    "optimistic"
+  );
+}
+
+const socialSet = (index) =>
+  Social.set(index, {
+    onCommit: () => {
+      onReport && onReport();
+    },
+    onCancel: () => onOpenChange(false),
+  });
+
+const reportAccount = () => {
+  socialSet({
+    index: {
+      flag: JSON.stringify({
+        key: "main",
+        value: contentModerationFlagValue,
+      }),
+    },
+  });
+};
+
+const reportAccountWithPosts = () => {
+  const flaggedAccountsList = getFlaggedAccountsList
+    ? JSON.parse(getFlaggedAccountsList)
+    : [];
+  const flaggedAccountsListSet = new Set(flaggedAccountsList);
+  const filterFlaggedAccounts = [
+    ...flaggedAccountsListSet.add(reportedAccountId),
+  ];
+
+  socialSet({
+    index: {
```

Comment:
This data won't go in the index. You can take out this line (and the matching brace.

Reply:
But why? We've been making a `Social.set()` all the time. @gabehamilton could you please provide more info here?

Reply:
Latest update. Let's use this format for compatibility with gateways that may want to use Social.index in their implementation.
```
    "index": {
      "moderation": `{\"key\":\"reported\",\"value\":{\"path\":\"${reportedAccountId}\", \"label\": \"report\"}}`
    },
    "moderation": {
      [reportedAccountId]: "report"
    },
```